### PR TITLE
refactor(map icons): use from `Icon` to be flexible

### PR DESCRIPTION
**Pull request by donni106**
- changed to render a map icon from our icon library instead of fixed location svg icons
- created `MapIcon` with default `'location'` as icon to be able to remove all occurrences across the code base where the location icon was set
- added `anchor` in `MapMarker` that was missing until now, to properly position icons on map with pin point at the bottom of location icons
  - removed icon anchor const exports as we do not need them individually
- added `OwnLocation` in our `Icon` library to be accessible with `Icon[_upperFirst(iconName)]`
- fixed type issues with `Icon[_upperFirst(iconName) as keyof typeof Icon]`

MQGB-34

### CODE REVIEW

File: src/components/map/Map.tsx [hunk 1]
```diff
@@ -25,6 +25,18 @@ type Props = {
...
+  iconName = 'location'
+}: {
+  iconColor?: string;
+  iconName?: string;
+}) => {
+  const MarkerIcon = Icon[_upperFirst(iconName) as keyof typeof Icon];
+
+  return <MarkerIcon color={iconColor} size={MARKER_ICON_SIZE} />;
+};
+
```
Comment (ardasnturk) [inline src/components/map/Map.tsx:39]: If the category does not have an icon, i.e. `iconName === null`, such an error occurs. Can we add an extra control here?


image
image

**Review by ardasnturk — CHANGES_REQUESTED** [review]
Why is the name of the Branch in German?

**Comment by donni106** [comment]
> Why is the name of the Branch in German?

Branch name is German because created directly from Jira, which is based on the ticket name.

File: src/components/map/Map.tsx (hunk 1 above)
Comment (donni106) [reply src/components/map/Map.tsx:39]: From which screen or component were you navigating? `'location'`. Where did you place the `console.log`? `iconName` should be string or undefined, to fall back to

Comment (ardasnturk) [reply src/components/map/Map.tsx:39]: route = POI -> Bildung -> Kartenansicht 

I placed the log in the `MapIcon`.
image

Comment (donni106) [reply src/components/map/Map.tsx:39]: I updated it to `iconName={marker.iconName ? marker.iconName : undefined}`

**Review by ardasnturk — APPROVED** [review]

**Review by d4v3d4vE — APPROVED** [review]

**Pull request merged**